misc/python/materialize/cli/ci_closed_issues_detect.py

```diff
@@ -0,0 +1,190 @@
+
+    response = requests.get(
+        f"https://api.github.com/repos/MaterializeInc/materialize/issues/{issue_id}",
+        headers=headers,
+    )
+
+    if response.status_code != 200:
+        raise ValueError(
+            f"Bad return code from GitHub: {response.status_code}: {response.text}"
+        )
+
+    issue_json = response.json()
+    assert (
+        issue_json["number"] == issue_id
+    ), f"Returned issue number {issue_json['number']} is not the expected issue number {issue_id}"
+    return issue_json["state"] == "closed"
+
+
+def filter_changed(issue_refs: list[IssueRef]) -> list[IssueRef]:
+    merge_base = buildkite.get_merge_base()
+    print(f"Merge base: {merge_base}")
+    result = spawn.capture(["git", "diff", "-U0", merge_base])
+    file_path = None
+    changed_lines = set()
+
+    for line in result.splitlines():
+        # +++ b/src/adapter/src/coord/command_handler.rs
+        if line.startswith("+++"):
+            file_path = line.removeprefix("+++ b/")
+        # @@ -641,7 +640,6 @@ impl Coordinator {
+        elif line.startswith("@@ "):
+            # We only care about the second value ("+640,6" in the example),
+            # which contains the line number and length of the modified block
+            # in new code state.
+            parts = line.split(" ")[2]
+            if "," in parts:
+                start, length = map(int, parts.split(","))
+            else:
+                start = int(parts)
+                length = 1
+            for line_nr in range(start, start + length):
+                changed_lines.add((file_path, line_nr))
+
+    return [
+        issue_ref
+        for issue_ref in issue_refs
+        if (issue_ref.filename, issue_ref.line_number) in changed_lines
+    ]
+
+
+def filter_closed(issue_refs: list[IssueRef]) -> list[IssueRef]:
+    issues = {issue_ref.issue_id for issue_ref in issue_refs}
+    closed_issues = {issue for issue in issues if is_issue_closed_on_github(issue)}
+    return [
+        issue_ref for issue_ref in issue_refs if issue_ref.issue_id in closed_issues
+    ]
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(
+        prog="ci-closed-issues-detect",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="ci-closed-issues-detect detects references to already closed GitHub issues.",
+    )
+
+    parser.add_argument(
+        "--changed-lines-only",
+        action="store_true",
+        help="only report issues in changed files/lines",
+    )
+
+    args = parser.parse_args()
+
+    filenames = spawn.capture(
+        ["git", "ls-tree", "--full-tree", "-r", "--name-only", "HEAD"]
+    )
+
+    issue_refs: list[IssueRef] = []
+    for filename in filenames.splitlines():
+        # Files without any ending can be interesting datadriven test files
+        if (
+            any([filename.endswith(ending) for ending in FILE_ENDINGS])
+            or "." not in filename
```

Comment:
This does not necessarily mean that a file has no ending; it could be a hidden file.

Why do we want to filter based on the file type at all? What do we gain from that? What do we not want to have included?